### ISSUE TITLE
PR approval process proposal

### ISSUE THREAD

**Issue by srtalbot** [cds-snc/platform-forms-client#3605]
**Problem to solve:** What security development standards should we enforce on Pull requests?

[Miro board for discussion](https://miro.com/app/board/uXjVKWHFPyw=/)

Update:
Here is the [Pull Request Process (Google doc)](https://docs.google.com/document/d/16sIDvlDKb-Qnifusm4qUbV0M45uFPeKfumOqTM07xAQ/)

**Comment by thiessenp-cds** [comment]
Hey @srtalbot here is the related Google Doc. Is there a good place to put this doc for discussion/future edits?

https://docs.google.com/document/d/1-Mo-LnY1L3VDKVckLShUJRkTdKSPO_8cZeq2sAv_748/edit#heading=h.kpg3p0hcoy2k

**Comment by srtalbot** [comment]
This is great, thanks for adding the document Pete. We can have the discussion in the doc.

**Comment by thiessenp-cds** [comment]
Here is an example case: [conditional input history for forms-form PR](https://github.com/cds-snc/platform-forms-client/pull/4036/)
![Screenshot 2024-08-01 at 12 23 29 PM](https://github.com/user-attachments/assets/a45ae5af-9c7d-4955-b92f-5c41d0acf66c)

How did this improve the pre-release?
- more eyes on the code to increase the possibility of finding any logical errors
- included developers not close to the feature that may catch errors that could be "invisible" to devs close to the feature
- included developers not close to the feature that will likely test the feature in unexpected ways outside the happy path e.g. Pat sure did :)
- increased overall confidence the feature was production ready

**Comment by Abi-Nada** [comment]
@thiessenp-cds to update the doc "draft the policy :D", and flag to devs to take a look.

**Comment by thiessenp-cds** [comment]
Here is the latest: [Security Development Operations - Pull Requests (Google doc)](https://docs.google.com/document/d/16sIDvlDKb-Qnifusm4qUbV0M45uFPeKfumOqTM07xAQ/edit#heading=h.kpg3p0hcoy2k)

**Comment by thiessenp-cds** [comment]
I updated the PR process from the feedback in the meeting held last week. I updated the status of the document to DRAFT and moved the [file into the shared forms google drive space](https://docs.google.com/document/d/16sIDvlDKb-Qnifusm4qUbV0M45uFPeKfumOqTM07xAQ/). I also left a [comment in the #forms-dev](https://gcdigital.slack.com/archives/C01GFD150MC/p1725546898995049) about the update and inviting people to comment/edit.

I think the devs have had enough time to comment/edit this document and this ticket can be closed. 

As a follow up idea, Anik mentioned using a GitHub template to help capture the PR process in a PR. This could work if devs are interested. (posted the question in #forms-dev)
